react-components/src/utilities/instanceIds/typeGuards.ts

```diff
@@ -7,18 +7,28 @@ import { type InstanceReference } from './types';
 
 export type AnnotationAssetRef = { id?: number; externalId?: string };
 
-export function isIdEither(instance: InstanceReference | AnnotationAssetRef): instance is IdEither {
+export function isIdEither(instance: unknown): instance is IdEither {
   return isExternalId(instance) || isInternalId(instance);
 }
 
-export function isExternalId(idEither: IdEither | AnnotationAssetRef): idEither is ExternalId {
-  return 'externalId' in idEither && !('space' in idEither);
+export function isExternalId(idEither: unknown): idEither is ExternalId {
+  return (
+    typeof idEither === 'object' &&
+    idEither !== null &&
+    'externalId' in idEither &&
+    !('space' in idEither)
```

Comment:
This guard is there, I am afraid of it being there. We should investigate if this can cause an error towards file or another object that can have space as part of its fields